src/join-managed.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+/// join-auth.sol -- Non-standard token adapters
+
+// Copyright (C) 2018 Rain <rainbreak@riseup.net>
+// Copyright (C) 2018-2020 Maker Ecosystem Growth Holdings, INC.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.6.12;
+
+interface VatLike {
+    function urns(bytes32, address) external view returns (uint256, uint256);
+    function dai(address) external view returns (uint256);
+    function gem(bytes32, address) external view returns (uint256);
```

Comment:
These 3 can be removed.

Reply:
We have these in cropJoin as well. They are used for testing (both local node and vs the mainnet contract). Would rather not diverge from having similar test code, unless you think it's important to remove them.

Reply:
I prefer to have special interfaces for testing if they are needed and keep the actual code that is going to Etherscan the cleaner as possible. In general I'd favor to extract whatever is needed for test in another `VatLike` interface which is there. Hadn't realized they were also in the other repo tbh.

Reply:
Done.